library/core/src/option.rs

```diff
@@ -985,6 +986,83 @@ impl<T> Option<T> {
     }
 }
 
+impl<T> Option<&T> {
+    /// Converts from `Option<&T>` (or `&Option<&T>`) to `*const T`.
+    ///
+    /// This is the opposite of `<*const T>::as_ref`.
```

Comment:
Can you turn `<*const T>::as_ref` into a link to its documentation?

Reply:
Sure thing. Do you happen to know how to intra-doc link pointers? Otherwise I'll just use the full URL.

Reply:
Paging @jyn514, who probably knows this. ^^

Reply:
`[..][pointer::as_ref]` seems to work, but that doesn't give you control over which of the two `as_ref`s it links to.

Reply:
Is it guaranteed that `#method.as_ref` and `#method.as_ref-1` will refer to the `const` and `mut` variants respectively? If not, it might be worth adding a custom `id`.

Reply:
`pointer` was recently implemented and isn't in beta yet, so you can't use it until the beta bump in 6 weeks (or else `doc --stage 0` will break). https://github.com/rust-lang/rust/pull/80181#issuecomment-748435901

There is currently no way to distinguish *const from *mut, I wouldn't consider `impl-1` stable and would prefer not to use it since the link checker won't warn if *const and *mut get swapped.